contracts/src/v0.8/dev/Flags.sol

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+
+import "../SimpleReadAccessController.sol";
+import "../interfaces/AccessControllerInterface.sol";
+import "../interfaces/FlagsDevInterface.sol";
+import "../interfaces/TypeAndVersionInterface.sol";
+
+
+/**
+ * @title The Flags contract
+ * @notice Allows flags to signal to any reader on the access control list.
+ * The owner can set flags, or designate other addresses to set flags. The
+ * owner must turn the flags off, other setters cannot. An expected pattern is
+ * to allow addresses to raise flags on themselves, so if you are subscribing to
+ * FlagOn events you should filter for addresses you care about.
+ */
+contract Flags is FlagsDevInterface, SimpleReadAccessController, TypeAndVersionInterface {
+
+  AccessControllerInterface public raisingAccessController;
+  AccessControllerInterface public loweringAccessController;
+
+  mapping(address => bool) private flags;
+
+  event FlagRaised(
+    address indexed subject
+  );
+  event FlagLowered(
+    address indexed subject
+  );
+  event RaisingAccessControllerUpdated(
+    address indexed previous,
+    address indexed current
+  );
+  event LoweringAccessControllerUpdated(
+    address indexed previous,
+    address indexed current
+  );
+
+  /**
+   * @param racAddress address for the raising access controller.
+   * @param lacAddress address for the lowering access controller.
+   */
+  constructor(
+    address racAddress,
+    address lacAddress
+  ) {
+    setRaisingAccessController(racAddress);
+    setLoweringAccessController(lacAddress);
+  }
+
+  function typeAndVersion()
+    external
+    pure
+    virtual
+    override
+    returns (
+      string memory
+    )
+  {
+    return "Flags 2.0.0";
+  }
+
+  /**
+   * @notice read the warning flag status of a contract address.
+   * @param subject The contract address being checked for a flag.
+   * @return A true value indicates that a flag was raised and a
+   * false value indicates that no flag was raised.
+   */
+  function getFlag(address subject)
```

Comment:
As we started to use flags for more than just tagging contracts, IMHO it makes sense to transition to a more general data type `bytes32`. cc @se3000 @kaleofduty wdyt?

Reply:
I'm not convinced. Is there a specific example use case where bytes32 works, but address does not?

In any case, I wouldn't make this change as part of this PR.

Reply:
Definitely, no making this change now, but something to consider for the future.